src/esp/assets/PhysicsManager.cpp

```diff
@@ -0,0 +1,266 @@
+  btGImpactCollisionAlgorithm::registerAlgorithm(_bDispatcher);
+  _bBroadphase = new btDbvtBroadphase();
+  _bSolver = new btSequentialImpulseConstraintSolver();
+  _bWorld = new btDiscreteDynamicsWorld(_bDispatcher, _bBroadphase, _bSolver,
+                                        _bCollisionConfig);
+
+  // btCollisionDispatcher *dispatcher = static_cast<btCollisionDispatcher
+  // *>(_bWorld.getDispatcher());
+  // btGImpactCollisionAlgorithm::registerAlgorithm(dispatcher);
+
+  _debugDraw.setMode(Magnum::BulletIntegration::DebugDraw::Mode::DrawWireframe);
+  _bWorld->setGravity({0.0f, -10.0f, 0.0f});
+  //_bWorld.setGravity({0.0f, 0.0f, -10.0f});
+  //_bWorld.setDebugDrawer(&_debugDraw);
+
+  physicsNode = node;
+
+  _timeline.start();
+  _initialized = true;
+  _do_profile = do_profile;
+  LOG(INFO) << "Initialized Physics Engine.";
+
+  return true;
+}
+
+PhysicsManager::~PhysicsManager() {
+  LOG(INFO) << "Deconstructing PhysicsManager";
+  if (_initialized) {
+    delete _bCollisionConfig;
+    delete _bDispatcher;
+    delete _bBroadphase;
+    delete _bSolver;
+    delete _bWorld;
+  }
+}
+
+void PhysicsManager::getPhysicsEngine() {}
+
+// Bullet Mesh conversion adapted from:
+// https://github.com/mosra/magnum-integration/issues/20
+bool PhysicsManager::initScene(
+    const AssetInfo& info,
+    const MeshMetaData& metaData,
+    std::vector<CollisionMeshData> meshGroup,
+    physics::BulletRigidObject* physObject) {
+
+  // Test Mesh primitive is valid
+  for (CollisionMeshData& meshData: meshGroup) {
+    if (!isMeshPrimitiveValid(meshData)) {return false;}
+  }
+
+  float mass = 0.0f;    // TODO (JH) weight is currently hardcoded
+  bool objectSuccess;
+  if (info.type == AssetType::INSTANCE_MESH) {
+    // ._semantic.ply mesh data
+    LOG(INFO) << "Initialize instance: before";
+    objectSuccess = physObject->initializeScene(info, mass, meshGroup, *_bWorld);
+    LOG(INFO) << "Initialize instance: after";
+    physObject->syncPose();
+  } 
+  else if (info.type == AssetType::FRL_INSTANCE_MESH) {
+    // FRL mesh
+    LOG(INFO) << "Initialize FRL: before";
+    objectSuccess = physObject->initializeScene(info, mass, meshGroup, *_bWorld);
+    LOG(INFO) << "Initialize FRL: after, success " << objectSuccess;
+    physObject->syncPose();
+  }
+  else {
+    // GLB mesh data
+    LOG(INFO) << "Initialize GLB: before";
+    objectSuccess = physObject->initializeScene(info, mass, meshGroup, *_bWorld);
+    LOG(INFO) << "Initialize GLB: after, success " <<objectSuccess;
+    physObject->syncPose();
+  }
+
+  LOG(INFO) << "Init scene done";
+  return objectSuccess;
+}
+
+
+bool PhysicsManager::initObject(
+    const AssetInfo& info,
+    const MeshMetaData& metaData,
+    std::vector<CollisionMeshData> meshGroup,
+    physics::BulletRigidObject* physObject) {
+
+  // Test Mesh primitive is valid
+  for (CollisionMeshData& meshData: meshGroup) {
+    if (!isMeshPrimitiveValid(meshData)) {return false;}
+  }
+
+  bool objectSuccess;
```

Comment:
Always initialize it.
objectSuccess = false; // or true, depending on your case